### PR TITLE
RFC: Adding a number of docker images

### PR DESCRIPTION
**Pull request by dellaert**
These were a number of github images I created to support assignments in a class, and I believe they might be valuable by themselves. Let's discuss whether this collection and structure for the docker folder is the right way to go.

---
This change is [Reviewable](https://reviewable.io/reviews/borglab/gtsam/241)

### CODE REVIEW

File: docker/ubuntu-gtsam-python/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# Get the base Ubuntu/GTSAM image from Docker Hub
+FROM dellaert/ubuntu-gtsam:bionic
```
Comment (varunagrawal) [inline docker/ubuntu-gtsam-python/Dockerfile]: Common namespace

File: docker/ubuntu-gtsam-python/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+# Install pip
+RUN apt-get install -y python-pip python-dev
```
Comment (varunagrawal) [inline docker/ubuntu-gtsam-python/Dockerfile]: This should say `python3-pip` and `python3-dev`. Ideally, we should be using `pyenv` to make python version management easy.

File: docker/ubuntu-gtsam-python/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+# Install python wrapper requirements
+RUN pip install -r /usr/src/gtsam/cython/requirements.txt
```
Comment (varunagrawal) [inline docker/ubuntu-gtsam-python/Dockerfile]: We need to check if `pip` invokes the built in `pip2` (which is deprecated) or `pip3`.

File: docker/ubuntu-gtsam/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+# Install git
+RUN apt-get update && \ 
+    apt-get install -y git
+
+# Install compiler
+RUN apt-get install -y build-essential
+
+# Clone GTSAM
+WORKDIR /usr/src/
+RUN git clone https://bitbucket.org/gtborg/gtsam.git
```
Comment (varunagrawal) [inline docker/ubuntu-gtsam/Dockerfile]: Needs to be updated to use `github` and perhaps pull the `develop` branch which has a lot of new niceness?

File: docker/ubuntu-gtsam/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,35 @@
+# Get the base Ubuntu image from Docker Hub
+FROM dellaert/ubuntu-boost-tbb-eigen3:bionic
+
+# Install git
+RUN apt-get update && \ 
+    apt-get install -y git
+
+# Install compiler
+RUN apt-get install -y build-essential
```
Comment (varunagrawal) [inline docker/ubuntu-gtsam/Dockerfile]: This and the previous command can be merged to finish in a single download.

File: docker/ubuntu-gtsam-python/Dockerfile (hunk 3 above)
Comment (jlblancoc) [reply docker/ubuntu-gtsam-python/Dockerfile]: what about using `python3 -m pip install ...` to be 100% sure?

Comment (varunagrawal) [reply docker/ubuntu-gtsam-python/Dockerfile]: That would be the best option. :slightly_smiling_face:

**Comment by dellaert** [comment]
@jlblancoc what's your take on this PR? :-)

**Comment by jlblancoc** [comment]
> @jlblancoc what's your take on this PR? :-)

Awesome, I made it work!

Screenshot from 2020-07-14 23-30-22

See the last commits, though, with a couple of details I needed to change to fix it. 
It was not obvious for me how to proceed, so wrote a short README.

**Comment by varunagrawal** [comment]
@dellaert I am deleting the branch `feature/docker_images` with the commit https://github.com/borglab/gtsam/commit/c1f9bc258bb62d1a9afdd96a5840f32abad06eb4 since it is a duplicate of this.

**Pull request merged**